src/components/content/credential/AddCredential.tsx

```diff
@@ -0,0 +1,271 @@
+        clear();
+        form.resetFields();
+    };
+
+    return (
+        <div className={'credential-from'}>
+            <Form
+                ref={formRef}
+                form={form}
+                labelCol={{ span: 4 }}
+                wrapperCol={{ span: 14 }}
+                layout='horizontal'
+                style={{ maxWidth: 1000 }}
+                onFinish={submit}
+            >
+                <div className={'credential-from-input'}>
+                    <Form.Item label='Name' name='name' rules={[{ required: true }]}>
+                        <Input />
+                    </Form.Item>
+
+                    <Form.Item label='Csp' name='csp' rules={[{ required: true, message: 'Please Select Csp!' }]}>
+                        <Select onSelect={handleCspSelect}>
+                            {Object.values(CredentialVariables.csp).map((csp: CredentialVariables.csp) => {
+                                return (
+                                    <Select.Option key={csp} value={csp}>
+                                        {csp}
+                                    </Select.Option>
+                                );
+                            })}
+                        </Select>
+                    </Form.Item>
+                    <Form.Item
+                        label='Type'
+                        name='type'
+                        rules={[{ required: true, message: 'Please Select The Type of Credential!' }]}
+                    >
+                        <Select onSelect={handleCredentialTypeSelect}>
+                            {/* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition */}
+                            {credentialTypeList && credentialTypeList.length > 0 ? (
+                                credentialTypeList.map((type: CredentialVariables.type) => {
+                                    return (
+                                        <Select.Option key={type} value={type}>
+                                            {type}
+                                        </Select.Option>
+                                    );
+                                })
+                            ) : (
+                                <></>
+                            )}
+                        </Select>
+                    </Form.Item>
+                    <Form.Item label='Description' name='description'>
+                        <TextArea rows={4} />
+                    </Form.Item>
+                    {/* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition */}
+                    {credentialVariableList && credentialVariableList.length > 0 ? (
+                        <Form.Item
+                            label='Variables'
+                            name='variables'
+                            rules={[
+                                {
+                                    required: requiredVariable === undefined ? false : requiredVariable,
```

Comment:
this logic is not correct. this is making all the parameters mandatory.
isMandatory field is there is each credentialVariable. We must use that.

Reply:
still mandatory variables not displayed